lib/rules/checked-requires-onchange-or-readonly.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @fileoverview Enforce the use of the 'onChange' or 'readonly' attribute when 'checked' is used'
+ * @author Jaesoekjjang
+ */
+
+'use strict';
+
+const ASTUtils = require('jsx-ast-utils');
+const isCreateElement = require('../util/isCreateElement');
+const report = require('../util/report');
+const docsUrl = require('../util/docsUrl');
+
+const messages = {
+  missingProperty: 'Checked should be used with either onChange or readOnly.',
+  exclusiveCheckedAttribute:
+    'Use either checked or defaultChecked but not both.',
+};
+
+const targetProps = ['checked', 'onChange', 'readOnly', 'defaultChecked'];
+const targetPropSet = new Set(targetProps);
+
+const defaultOptions = {
+  ignoreMissingProperties: true,
+  ignoreExclusiveCheckedAttribute: true,
+};
+
+function extractTargetProps(properties, keyName) {
+  return properties.reduce((acc, prop) => {
+    if (
+      prop[keyName]
+      && prop[keyName].name
+      && targetPropSet.has(prop[keyName].name)
```

Comment:
```suggestion
      && targetPropSet.has(prop[keyName].name)
```
since if it's falsy or undefined it won't be in the set anyways